bitmath/__init__.py

```diff
@@ -413,6 +413,19 @@ def __str__(self):
         global format_string
         return self.format(format_string)
 
+    def __format__(self, format_spec):
```

Comment:
For the purpose of updating the documentation later, can you tell me some of the limitations of this? Additional more complex examples would be even better. Even just a gist w/ you playing around with this in a python interpreter showing example input and output would be helpful. Like, as a user, still you use the various [instance attributes](http://bitmath.readthedocs.io/en/latest/instances.html#instances-attributes) in formatting? I *think* so, since the `__format__` method ultimately calls out to `self.format`, but I'd like some more examples and such :smile: